featuretools/entityset/entityset.py

```diff
@@ -1008,66 +1006,74 @@ def add_last_time_indexes(self, updated_dataframes=None):
     # #  Other ###############################################
     # ###########################################################################
 
-    # def add_interesting_values(self, max_values=5, verbose=False, entity_id=None):
-    #     """Find interesting values for categorical variables, to be used to generate "where" clauses
+    def add_interesting_values(self, max_values=5, verbose=False, dataframe_id=None, values=None):
+        """Find interesting values for categorical columns, to be used to generate "where" clauses
 
-    #     Args:
-    #         max_values (int) : Maximum number of values per variable to add.
-    #         verbose (bool) : If True, print summary of interesting values found.
-    #         entity_id (str) : The Entity for which to add interesting values. If not specified
-    #             interesting values will be added for all Entities.
+        Args:
+            max_values (int) : Maximum number of values per column to add.
+            verbose (bool) : If True, print summary of interesting values found.
+            dataframe_id (str) : The dataframe in the EntitySet for which to add interesting values.
+                If not specified interesting values will be added for all dataframes.
+            values (dict): A dictionary mapping column names to the interesting values to set
```

Comment:
Is the purpose of adding a `values` parameter to give users the ability to avoid having to bring dask and koalas dataframes into memory? 

If so, maybe we should add a note in the docstring about it being useful for large datasets. 

The other things is that unless we're going to explicitly disallow finding the interesting values for dask and koalas dataframes, we should probably include them in the tests in some way, even if it's just xfailing and saying that we're recommending it not be used.

Reply:
Previously there was a setter on the Variable that users could use to manually assign specific interesting values, but that no longer will exist after the Woodwork integration. The `values` parameter here is what is providing that same type of functionality.

We currently do not support adding interesting values automatically for Dask or Koalas as that would require pulling the data into memory, so Dask/Koalas users must pass via the `values` argument. This approach is tested in `test_add_interesting_values_specified_vals`. 

When we originally did the Dask and Koalas implementation for Featuretools we adopted the convention that if we ran a test only for pandas, that means we are not supporting that behavior for Dask and Koalas, and don't have plans to support, but if we included an xfail that was a feature that we don't currently support but would like to. The idea was that the xfails would point us to things to work on. I'm not sure we have followed that strictly though - might be worth a conversation to determine how we want to handle moving forward since we haven't used that approach in other libraries that I'm aware of.

Reply:
Ah, I see. Thank you for the explanation! (I definitely have not been consciously following the xfail convention in woodwork, but it seems like a good one to have!)

I guess my question, then, becomes whether we should tell users in the docstring that they should supply `values` for dask and koalas dataframes. If I'm understanding correctly, though, it seems like featuretools assumes a user working with dask or koalas has read through the corresponding guides and knows about all the limitations through that. In which case, we should leave any additional notes about the limitations to the guides.